src/generics/middleware/checkAdminRole.js

```diff
@@ -0,0 +1,151 @@
+
+				// Decode the JWT to extract its claims without verifying
+				const tokenClaims = jwt.decode(token, { complete: true })
+
+				if (!tokenClaims || !tokenClaims.header) {
+					// If the token does not contain valid claims or header, send an unauthorized response
+					rspObj.errCode = CONSTANTS.apiResponses.TOKEN_MISSING_CODE
+					rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_MISSING_MESSAGE
+					rspObj.responseCode = HTTP_STATUS_CODE['unauthorized'].status
+					return res.status(HTTP_STATUS_CODE['unauthorized'].status).send(respUtil(rspObj))
+				}
+
+				// Extract the key ID (kid) from the token header
+				const kid = tokenClaims.header.kid
+
+				// Construct the path to the public key file using the key ID
+				let filePath = path.resolve(__dirname, keycloakPublicKeyPath, kid.replace(/\.\.\//g, ''))
+
+				// Read the public key file from the resolved file path
+				const accessKeyFile = await fs.promises.readFile(filePath, 'utf8')
+
+				// Ensure the public key is properly formatted with BEGIN and END markers
+				const cert = accessKeyFile.includes(PEM_FILE_BEGIN_STRING)
+					? accessKeyFile
+					: `${PEM_FILE_BEGIN_STRING}\n${accessKeyFile}\n${PEM_FILE_END_STRING}`
+				let verifiedClaims
+				try {
+					// Verify the JWT using the public key and specified algorithms
+					verifiedClaims = jwt.verify(token, cert, { algorithms: ['sha1', 'RS256', 'HS256'] })
+				} catch (err) {
+					// If the token is expired or any other error occurs during verification
+					if (err.name === 'TokenExpiredError') {
+						rspObj.errCode = CONSTANTS.apiResponses.TOKEN_INVALID_CODE
+						rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_INVALID_MESSAGE
+						rspObj.responseCode = HTTP_STATUS_CODE['unauthorized'].status
+						return res.status(HTTP_STATUS_CODE['unauthorized'].status).send(respUtil(rspObj))
+					}
+				}
+
+				// Extract the external user ID from the verified claims
+				const externalUserId = verifiedClaims.sub.split(':').pop()
+
+				const data = {
+					id: externalUserId,
+					roles: [], // this is temporariy set to an empty array, it will be corrected soon...
+					name: verifiedClaims.name,
+					organization_id: verifiedClaims.org || null,
+				}
+
+				// Ensure decodedToken is initialized as an object
+				decodedToken = decodedToken || {}
+				decodedToken['data'] = data
+			}
+		} catch (error) {
+			return res.status(HTTP_STATUS_CODE.unauthorized.status).send(
+				respUtil({
+					errCode: CONSTANTS.apiResponses.TOKEN_MISSING_CODE,
+					errMsg: CONSTANTS.apiResponses.TOKEN_MISSING_MESSAGE,
+					responseCode: HTTP_STATUS_CODE['unauthorized'].status,
+				})
+			)
+		}
+
+		if (!decodedToken.data?.organizations?.[0]?.roles) {
+			return next({
+				status: HTTP_STATUS_CODE.forbidden.status,
+				message: CONSTANTS.apiResponses.TOKEN_INVALID_CODE,
+			})
+		}
+
+		// Extract roles from decoded token payload
+		let fetchRoleFromToken = decodedToken.data.organizations[0].roles
```

Comment:
@MallanagoudaB can we make this configurable
by default keep the existing logic. if path.resolve(__dirname, '../../config.json') has file and key value exists use it
{
    "authTokenUserInformation": {
        "userId": "data.id",
        "userName": "data.name",
        "firstName": "data.name",
        "organization_id": "data.organization_codes[0]",
        "roles": "data.organizations[?code={{organization_id}}].roles",
        "organizations": "data.organizations",
        "tenant_id": "data.tenant_code"
    },
    userRolesInformationKey : // they key where we check roles array
}